app/exporter/logic/exports.py

```diff
@@ -0,0 +1,89 @@
+"""
+    This module contains all business logic around exports, including
+    validating ownership, creating new models, and processing pending
+    exports
+"""
+import os
+import csv
+from uuid import uuid4
+
+from exporter.db import exports
+from exporter.models.export import Export
+from exporter.logic import mailgun
+from exporter.logic import s3
+from flask import abort
+
+
+def get(user, id):
+    export = exports.find_one(Export.key(user, id))
+    return Export(export)
+
+
+def list_exports(api_key, domain):
+    export_list = exports.find({
+        'domain': domain,
+        'user': api_key
+    })
+    return [Export(e) for e in export_list]
+
+
+def create(api_key, domain, export_type):
+
+    if export_type not in Export.TYPES:
+        abort(400)
+
+    new_export = Export({
+        "domain": domain,
+        "user": api_key,
+        "type": export_type,
+        "status": "pending",
+        "id": str(uuid4())
+    })
+
+    exports.insert_one(new_export.serialize())
+
+    return new_export
+
+
+def process_pending():
+    for e in exports.find({'status': "pending"}):
```

Comment:
This may take a while especially if there are several pending jobs for domains that have a lot of suppressions. Have you thought about parallelizing this?